test/e2e/common/secrets.go

```diff
@@ -134,6 +136,66 @@ var _ = ginkgo.Describe("[sig-api-machinery] Secrets", func() {
 		secret, err := createEmptyKeySecretForTest(f)
 		framework.ExpectError(err, "created secret %q with empty key in namespace %q", secret.Name, f.Namespace.Name)
 	})
+
+	ginkgo.It("should patch a secret", func() {
+		ginkgo.By("creating a secret")
+
+		secretTestName := "test-secret-" + string(uuid.NewUUID())
+
+		// create a secret in namespace default
+		_, err := f.ClientSet.CoreV1().Secrets("default").Create(&v1.Secret{
+			ObjectMeta: metav1.ObjectMeta{
+				Name: secretTestName,
+			},
+			Data: map[string][]byte{
+				"key": []byte("value"),
+			},
+			Type: "Opaque",
+		})
+		framework.ExpectNoError(err, "failed to create secret")
+
+		ginkgo.By("listing secrets in all namespaces to ensure that there are more than zero")
+		// list all secrets in namespace default
+		secretsList, err := f.ClientSet.CoreV1().Secrets("").List(metav1.ListOptions{})
+		framework.ExpectNoError(err, "failed to list secrets")
+		framework.ExpectNotEqual(len(secretsList.Items), 0, "no secrets found")
+
+		foundCreatedSecret := false
+		var secretCreatedName string
+		for _, val := range secretsList.Items {
+			if val.ObjectMeta.Name == secretTestName && string(val.Data["key"]) == "value" {
+				foundCreatedSecret = true
+				secretCreatedName = val.ObjectMeta.Name
+			}
+		}
+		framework.ExpectEqual(foundCreatedSecret, true, "unable to find secret by its value")
+
+		ginkgo.By("patching the secret")
+		// patch the secret
+		secretPatchNewData := base64.StdEncoding.EncodeToString([]byte("value1"))
+		secretPatch := fmt.Sprintf(`{"metadata":{"labels":{"testsecret":"true"}},"data":{"key":"%v"}}`, secretPatchNewData)
```

Comment:
don't generate json with Sprintf, serialize a struct, e.g.:

```go
secretPatch, err := json.Marshal(map[string]interface{}{
  "metadata": map[string]interface{}{
    "labels": map[string]string{"testsecret":"true"},
  },
  "data": map[string][]byte{"key": []byte("value1")},
})
```

Reply:
Updated in b4fb46409d1